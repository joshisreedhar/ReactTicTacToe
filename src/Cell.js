import React from 'react';
import styled from 'styled-components';

const Div= styled.div`
border: 1px;
border-color:black;
`;


class Cell extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            cellText: props.cellText ? props.cellText : ""
        };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        if(this.state.cellText === ""){
            this.setState({cellText:"x"});
            this.props.onSelect();
        }
    }

    render(){
        return <Div 
                id={`cell${this.props.row}${this.props.col}`} 
                onClick={this.handleClick}>{this.state.cellText}</Div>
    }
}

export default Cell;

