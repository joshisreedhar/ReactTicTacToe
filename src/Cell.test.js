import React from 'react';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Cell from './Cell';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';




it('Cell should render', ()=>{
    const cell = renderer
        .create(<Cell cellText={"test"} col="1" row ="1"/>)
        .toJSON();

    expect(cell).toMatchSnapshot();
});

it('Empty Cell should disptch call when clicked', ()=>{
    Enzyme.configure({ adapter: new Adapter() })
    const mockHandler = jest.fn();
    let wrapper = shallow(<Cell row="1" col="1" cellText="" onSelect={mockHandler} />);
    wrapper.find('#cell11').simulate('click');
    expect(mockHandler.mock.calls.length).toBe(1);
});

it('Filled Cell should not disptch call when clicked', ()=>{
    Enzyme.configure({ adapter: new Adapter() })
    const mockHandler = jest.fn();
    let wrapper = shallow(<Cell row="1" col="1" cellText="X" onSelect={mockHandler} />);
    wrapper.find('#cell11').simulate('click');
    expect(mockHandler.mock.calls.length).toBe(0);
});

it('Empty cell should be filled when clicked', ()=>{
    Enzyme.configure({ adapter: new Adapter() })
    const mockHandler = jest.fn();
    let wrapper = shallow(<Cell row="1" col="1" cellText="" onSelect={mockHandler} />);
    wrapper.find('#cell11').simulate('click');
    expect(wrapper.state().cellText).toBe("x");
});