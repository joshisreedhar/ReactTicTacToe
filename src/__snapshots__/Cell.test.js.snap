// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should render 1`] = `
<div
  className="sc-bdVaJa TPpzq"
  id="cell11"
  onClick={[Function]}
>
  test
</div>
`;
